operator/azure.go

```diff
@@ -56,12 +56,12 @@ func startAzureAllocator(clientQPSLimit float64, clientBurst int) error {
 	azureClient, err := azureAPI.NewClient(option.Config.AzureSubscriptionID,
 		option.Config.AzureResourceGroup, azMetrics, clientQPSLimit, clientBurst)
 	if err != nil {
-		return fmt.Errorf("unable to create Azure client: %s", err)
+		return nil, fmt.Errorf("unable to create Azure client: %s", err)
 	}
 	instances := azureIPAM.NewInstancesManager(azureClient)
-	nodeManager, err = ipam.NewNodeManager(instances, &ciliumNodeUpdateImplementation{}, iMetrics, option.Config.ParallelAllocWorkers, false)
+	nodeManager, err := ipam.NewNodeManager(instances, &ciliumNodeUpdateImplementation{}, iMetrics, option.Config.ParallelAllocWorkers, false)
 	if err != nil {
-		return fmt.Errorf("unable to initialize Azure node manager: %s", err)
+		return nil, fmt.Errorf("unable to initialize Azure node manager: %s", err)
 	}
 
 	instances.Resync(context.TODO())
```

Comment:
Do we want this and the goroutine to remain under `operator` (either in main package or a sub-package), or it would be fine to move somewhere into `pkg/`? I can imagine it maybe a convention where it's desirable for anything that uses log package and starts go routines to be out of `pkg/`, but no idea if that applies or I'm overthinking it.

E.g. there `pkg/azure/ipam` and  `pkg/ipam/allocator`, but for some reason this function lives here, so does the AWS conterpart. Would it better to move these into `pkg/` or something like `operator/tasks/{azure,aws}_allocator`?

Reply:
Good point, the code is basically the same for all IPAM modes at this point. I'll move it into the generic ipam package.

Reply:
Done